src/servicex/servicex.py

```diff
@@ -36,25 +36,42 @@
 import os
 import getpass
 
-parser = argparse.ArgumentParser("ServiceX Command Line Interface")
-parser.add_argument("command", choices=["init", "version", "clear"])
-parser.add_argument("--cert-dir",
-                    default="~/.globus",
-                    help="Directory where your grid certs are located")
-parser.add_argument("--namespace", "-n",
-                    default='default',
-                    help="Namespace where secret should be created")
-
-
-def init_cluster(cert_dir, secret_name, namespace):
+parser = argparse.ArgumentParser("servicex")
+parser.add_argument('--namespace', "-n",
+                    default="default",
+                    help="Namespace where secrets should be created")
+subparsers = parser.add_subparsers(dest="command", required=True)
```

Comment:
It looks like `required` argument was added in 3.7
Let's keep this compatible with 3.6 and above like the setup.py states.

Reply:
It looks like the following worked as of 3.6:
```
subparsers = parser.add_subparsers(dest="command")
subparsers.required = True
```
This is because `add_subparsers` returns an `Action` object, which has a `required` attribute. The `required` argument was added for convenience in 3.7. Updated with the above.

Reply:
The sub parsers are very picky - please update the README to reflect that the --namspace argument has to come before the command:
```
servicex init certs --namespace foo
usage: servicex [-h] [--namespace NAMESPACE] {init,clear,version} ...
servicex: error: unrecognized arguments: --namespace foo
```